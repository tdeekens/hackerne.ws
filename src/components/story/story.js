import React from 'react';
import PropTypes from 'prop-types';
import styles from './story.module.css';

const Link = props => (
  <div className={styles.link}>
    <a href={props.url}>
      <div className={styles.title}>{props.title}</div>
    </a>
  </div>
);
Link.displayName = 'Link';
Link.propTypes = {
  title: PropTypes.string.isRequired,
};

const PublishedAt = props => {
  const options = {
    weekday: 'long',
    year: 'numeric',
    month: 'long',
    day: 'numeric',
    hour: 'numeric',
    minute: 'numeric',
  };

  return (
    <div className={styles.publishedAt}>
      at{' '}
      {new Date(props.timestamp * 1000).toLocaleDateString(
        props.locale,
        options
      )}
    </div>
  );
};
PublishedAt.displayName = 'PublishedAt';
PublishedAt.propTypes = {
  locale: PropTypes.string.isRequired,
  timestamp: PropTypes.number.isRequired,
};
PublishedAt.defaultProps = {
  locale: 'en-US',
};

export const Story = props => (
  <div className={styles.story}>
    <div className={styles.score}>{props.score}</div>
    <div>
      <Link url={props.url} title={props.title} />
      <div className={styles.meta}>
        <div className={styles.author}>by {props.author}</div>
        <PublishedAt timestamp={props.publishedAt} />
      </div>
    </div>
  </div>
);
Story.displayName = 'Story';
Story.propTypes = {
  id: PropTypes.number.isRequired,
  author: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  url: PropTypes.string,
  score: PropTypes.number.isRequired,
  publishedAt: PropTypes.number.isRequired,
};

export default Story;
