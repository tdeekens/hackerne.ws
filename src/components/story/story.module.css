@keyframes fadein {
   from { opacity: 0; }
   to   { opacity: 1; }
}

.story {
   display: flex;
   flex-direction: row;
   align-items: center;
   background-color: #fff;
   width: 100%;
   animation: fadein 0.3s;
}

.story > * + * {
   margin: 1.5em 0;
}

.score {
   color: #f60;
   font-size: 16px;
   font-weight: bold;
   margin: 12px
}

.information {
   display: flex;
   flex-direction: row;
}

.information > * + * {
   margin: 0 4px;
}

.title {
   color: #000;
   font-size: 14px;
}

.link {
   color: #828282;
   font-size: 14px;
}

.link > a{
   text-decoration: none;
}

.meta {
   display: flex;
   flex-direction: row;
}

.meta > * + * {
   margin: 0 4px;
}

.author, .publishedAt {
   color: #828282;
   font-size: 14px;
}
