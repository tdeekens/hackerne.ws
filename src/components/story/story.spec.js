import React from 'react';
import { shallow } from 'enzyme';
import { Story } from './story';

const createTestProps = customProps => ({
  id: 1,
  author: 'John Doe',
  title: 'Test story title',
  url: 'http://test-url.com',
  score: 12,
  publishedAt: 1541410855801,
  ...customProps,
});

describe('rendering', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = createTestProps();
    wrapper = shallow(<Story {...props} />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the author', () => {
    expect(wrapper).toIncludeText(props.author);
  });
});
