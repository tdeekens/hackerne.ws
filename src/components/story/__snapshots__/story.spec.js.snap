// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should match snapshot 1`] = `
<div>
  <div>
    12
  </div>
  <div>
    <Link
      title="Test story title"
      url="http://test-url.com"
    />
    <div>
      <div>
        by 
        John Doe
      </div>
      <PublishedAt
        locale="en-US"
        timestamp={1541410855801}
      />
    </div>
  </div>
</div>
`;
